### PR TITLE
Add sample for writing messages from Solace to BigQuery

### PR DESCRIPTION
**Pull request by himoacs**

### CODE REVIEW

**Comment by Nephery** [comment]
Thanks @himoacs . Could you also update the [Samples section in the README](https://github.com/himoacs/solace-apache-beam/blob/master/README.md#sample-walkthrough) with what's needed to run this:

* Add a subsection for this sample in the [Run a Sample section](https://github.com/himoacs/solace-apache-beam/blob/master/README.md#run-a-sample). Within which, could you write what exactly needs to be setup in BigQuery, as well as the command needed to run the sample itself?
* Also, seeing as how there's a pre-defined payload for this sample, I think we'll need to add a little blurb in [Populate the Solace PubSub+ Queues section](https://github.com/himoacs/solace-apache-beam/blob/master/README.md#populate-the-solace-pubsub-queues). Something to say to publish messages in this format instead of doing a `-pfl=README.md`.

**Comment by himoacs** [comment]
Thanks, I have added documentation for the new sample.

File: README.md [hunk 1]
```diff
@@ -214,6 +220,27 @@ The [SolaceProtoBuffRecordTest](solace-apache-beam-samples/src/main/java/com/sol
     ```
 1. Verify that the messages are being outputted in the log.
 
+#### SolaceBigQuery
+
+The [SolaceBigQuery](solace-apache-beam-samples/src/main/java/com/solace/connector/beam/examples/SolaceBigQuery.java) example receives a message from PubSub+ queue, transforms it, and then writes it to a BigQuery table. 
+
+1. Create a BigQuery table with appropriate schema.
```
Comment (Nephery) [inline README.md:227]: Could you describe what exactly is the "appropriate schema"? This is to help people reading the document to easily run the sample without needing to dig around too much.

File: README.md [hunk 2]
```diff
@@ -214,6 +220,27 @@ The [SolaceProtoBuffRecordTest](solace-apache-beam-samples/src/main/java/com/sol
...
+1. Modify SolaceBigQuery sample to add details about your table (project id, dataset id, and table id) in the following section:
+```java
+TableReference tableSpec =
+                new TableReference()
+                        .setProjectId("<project_id>")
+                        .setDatasetId("<dataset_id>")
+                        .setTableId("<table_id>");
+```
```
Comment (Nephery) [inline README.md:235]: Could we add the properties to the sample's `Options` interface?

For example:
- `bigqueryProject` for `<project_id>`
- `bigqueryDataset` for `<dataset_id>`
- `bigqueryTable` for `<table_id>`

File: README.md [hunk 3]
```diff
@@ -214,6 +220,112 @@ The [SolaceProtoBuffRecordTest](solace-apache-beam-samples/src/main/java/com/sol
...
+
+1. Run the SolaceBigQuery sample on Google Dataflow:
+```shell script
+mvn compile exec:java -Dexec.mainClass=com.solace.connector.beam.examples.SolaceBeamBigQuery -Dexec.args="--sql=Q/fx --cip=${SOLACE_URI} --cu=${SOLACE_USERNAME} --cp=${SOLACE_PASSWORD} --vpn=${SOLACE_VPN} --project=<project_name> --tempLocation=gs://<bucket_name>/demo --workerMachineType=n1-standard-2 --runner=DataflowRunner --autoscalingAlgorithm=THROUGHPUT_BASED --maxNumWorkers=4 --stagingLocation=gs://<bucket_name>/staging" -Pdataflow-runner
```
Comment (Nephery) [inline README.md:324]: - `--sql=Q/fx` should be `--sql=Q/fx-001`
- `--project=<project_name>` should be `--project=${GCP_PROJECT}`
- `--tempLocation=gs://<bucket_name>/demo` should be `--tempLocation=${GOOGLE_STORAGE_TMP}`
- `--stagingLocation=gs://<bucket_name>/staging` should be `--stagingLocation=${GOOGLE_STORAGE_STAGING}`
- Add `--bigQueryProject=${BIGQUERY_PROJECT}`
- Add `--bigQueryDataset=${BIGQUERY_DATASET}`
- Add `--bigQueryTable=${BIGQUERY_TABLE}`

File: README.md [hunk 4]
```diff
@@ -214,6 +220,103 @@ The [SolaceProtoBuffRecordTest](solace-apache-beam-samples/src/main/java/com/sol
...
+ {
+   "name": "vwap",
+   "type": "FLOAT",
+   "mode": "NULLABLE"
+ }
+]
+```
+1. Run the SolaceBigQuery sample on Google Dataflow:
+```shell script
+mvn compile exec:java -Dexec.mainClass=com.solace.connector.beam.examples.SolaceBeamBigQuery -Dexec.args="--sql=Q/fx-001 --cip=${SOLACE_URI} --cu=${SOLACE_USERNAME} --cp=${SOLACE_PASSWORD} --vpn=${SOLACE_VPN} --project=${GCP_PROJECT} --bigQueryProject=${BIGQUERY_PROJECT} --bigQueryDataset=${BIGQUERY_DATASET} --bigQueryTable=${BIGQUERY_TABLE} --tempLocation=${GOOGLE_STORAGE_TMP} --workerMachineType=n1-standard-2 --runner=DataflowRunner --autoscalingAlgorithm=THROUGHPUT_BASED --maxNumWorkers=4 --stagingLocation=${GOOGLE_STORAGE_STAGING}" -Pdataflow-runner
```
Comment (Nephery) [inline README.md:315]: `-Dexec.mainClass=com.solace.connector.beam.examples.SolaceBeamBigQuery` needs to be -`Dexec.mainClass=com.solace.connector.beam.examples.SolaceBigQuery`

File: solace-apache-beam-samples/src/main/java/com/solace/connector/beam/examples/SolaceBigQuery.java [hunk 5]
```diff
@@ -0,0 +1,245 @@
...
+
+		void setSmi(boolean value);
+
+		@Description("The timeout in milliseconds while try to receive a messages from Solace broker")
+		@Default.Integer(100)
+		int getTimeout();
+
+		void setTimeout(int timeoutInMillis);
+
+		@Validation.Required
```
Comment (Nephery) [inline solace-apache-beam-samples/src/main/java/com/solace/connector/beam/examples/SolaceBigQuery.java:110]: `Validation` needs to be imported.

**Comment by Nephery** [comment]
Thanks for the fixes @himoacs. This is good to merge.

**Pull request merged**